### ISSUE TITLE
ClassCastException when MessageHandler is added to Session as a "Lambda" expression

### ISSUE THREAD

**Issue by asgs** [tyrus-project/tyrus#620]
in the `onOpen()` method of a custom `Endpoint` class, when a `MessageHandler.Whole<String>` is registered to the session as a simple Anonymous class, it works without any problem. But when it's converted to a lamdba, the following error occurs.

``java.lang.ClassCastException: java.base/java.lang.Boolean cannot be cast to java.base/java.lang.String
	at org.glassfish.tyrus.core.TyrusSession.notifyMessageHandlers(TyrusSession.java:576)
	at org.glassfish.tyrus.core.TyrusEndpointWrapper.onMessage(TyrusEndpointWrapper.java:879)
	at org.glassfish.tyrus.core.TyrusWebSocket.onMessage(TyrusWebSocket.java:216)
	at org.glassfish.tyrus.core.frame.TextFrame.respond(TextFrame.java:139)
	at org.glassfish.tyrus.core.ProtocolHandler.process(ProtocolHandler.java:807)
	at org.glassfish.tyrus.core.TyrusWebSocketEngine$TyrusReadHandler.handle(TyrusWebSocketEngine.java:562)
	at org.glassfish.tyrus.container.grizzly.server.GrizzlyServerFilter$ProcessTask.execute(GrizzlyServerFilter.java:379)
	at org.glassfish.tyrus.container.grizzly.client.TaskProcessor.processTask(TaskProcessor.java:114)
	at org.glassfish.tyrus.container.grizzly.client.TaskProcessor.processTask(TaskProcessor.java:91)
	at org.glassfish.tyrus.container.grizzly.server.GrizzlyServerFilter.handleRead(GrizzlyServerFilter.java:215)
	at org.glassfish.grizzly.filterchain.ExecutorResolver$9.execute(ExecutorResolver.java:119)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeFilter(DefaultFilterChain.java:284)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeChainPart(DefaultFilterChain.java:201)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.execute(DefaultFilterChain.java:133)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.process(DefaultFilterChain.java:112)
	at org.glassfish.grizzly.ProcessorExecutor.execute(ProcessorExecutor.java:77)
	at org.glassfish.grizzly.nio.transport.TCPNIOTransport.fireIOEvent(TCPNIOTransport.java:526)
	at org.glassfish.grizzly.strategies.AbstractIOStrategy.fireIOEvent(AbstractIOStrategy.java:112)
	at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy.run0(WorkerThreadIOStrategy.java:117)
	at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy.access$100(WorkerThreadIOStrategy.java:56)
	at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy$WorkerThreadRunnable.run(WorkerThreadIOStrategy.java:137)
	at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.doWork(AbstractThreadPool.java:591)
	at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.run(AbstractThreadPool.java:571)
	at java.base/java.lang.Thread.run(Thread.java:844)``

Looking at the [Tyrus src code](https://github.com/tyrus-project/tyrus/blob/master/core/src/main/java/org/glassfish/tyrus/core/TyrusSession.java#L576) where this error occurs, I wonder if it's an incorrect or unsafe cast that is causing this.

The code which I tried to convert to lamdba is at [JShellWebSocketEndpoint.java](https://github.com/asgs/jshell-connector/blob/master/src/main/java/org/asgs/jshellconnector/websocket/JShellWebSocketEndpoint.java#L33). I wonder why it only happens with a Lambda expression.

The exact Lamdba snippet IntelliJ offered is below.

```session.addMessageHandler((MessageHandler.Whole<String>) message -> System.out.println(message));```

**Comment by pavelbucek** [comment]
Ah, I'm sorry, I did not know that the formatting on the blog is messed up.

TLDR; please use Session.addMessageHandler(Class<T>, 
MessageHandler.Whole<T>); [1] version without a class can't be used with 
lambdas (which is described in javadoc).

Regards,
Pavel

[1] 
https://tyrus-project.github.io/apidocs-javax.websocket/1.1/javax/websocket/Session.html#addMessageHandler(java.lang.Class,%20javax.websocket.MessageHandler.Whole)

**Comment by pavelbucek** [comment]
Please see https://blogs.oracle.com/pavelbucek/websocket-api-11-released

**Comment by asgs** [comment]
Thanks, worked perfectly fine!
